osf/external/gravy_valet/request_helpers.py

```diff
@@ -0,0 +1,229 @@
+
+ACCOUNT_EXTERNAL_SERVICE_PATH = 'external_storage_service'
+ADDON_EXTERNAL_SERVICE_PATH = 'base_account.external_storage_service'
+
+
+def get_account(gv_account_pk, requesting_user):  # -> JSONAPIResultEntry
+    '''Return a JSONAPIResultEntry representing a known AuthorizedStorageAccount.'''
+    return get_gv_result(
+        endpoint_url=ACCOUNT_ENDPOINT.format(pk=gv_account_pk),
+        requesting_user=requesting_user,
+        params={'include': ACCOUNT_EXTERNAL_SERVICE_PATH},
+    )
+
+
+def get_addon(gv_addon_pk, requested_resource, requesting_user):  # -> JSONAPIResultEntry
+    '''Return a JSONAPIResultEntry representing a known ConfiguredStorageAddon.'''
+    return get_gv_result(
+        endpoint_url=ADDON_ENDPOINT.format(pk=gv_addon_pk),
+        requesting_user=requesting_user,
+        requested_resource=requested_resource,
+        params={'include': ADDON_EXTERNAL_SERVICE_PATH},
+    )
+
+
+def iterate_accounts_for_user(requesting_user):  # -> typing.Iterator[JSONAPIResultEntry]
+    '''Returns an iterator of JSONAPIResultEntries representing all of the AuthorizedStorageAccounts for a user.'''
+    user_result = get_gv_result(
+        endpoint_url=USER_LIST_ENDPOINT,
+        requesting_user=requesting_user,
+        params={'filter[user_uri]': requesting_user.get_semantic_iri()},
+    )
+    if not user_result:
+        return None
+    yield from iterate_gv_results(
+        endpoint_url=user_result.get_related_link('authorized_storage_accounts'),
+        requesting_user=requesting_user,
+        params={'include': ACCOUNT_EXTERNAL_SERVICE_PATH},
+    )
+
+
+def iterate_addons_for_resource(requested_resource, requesting_user):  # -> typing.Iterator[JSONAPIResultEntry]
+    '''Returns an iterator of JSONAPIResultEntires representing all of the ConfiguredStorageAddons for a resource.'''
+    resource_result = get_gv_result(
+        endpoint_url=RESOURCE_LIST_ENDPOINT,
+        requesting_user=requesting_user,
+        requested_resource=requested_resource,
+        params={'filter[resource_uri]': requested_resource.get_semantic_iri()},
+    )
+    if not resource_result:
+        return None
+    yield from iterate_gv_results(
+        endpoint_url=resource_result.get_related_link('configured_storage_addons'),
+        requesting_user=requesting_user,
+        requested_resource=requested_resource,
+        params={'include': ADDON_EXTERNAL_SERVICE_PATH}
+    )
+
+
+def get_waterbutler_config(gv_addon_pk, requested_resource, requesting_user):  # -> JSONAPIResultEntry
+    return get_gv_result(
+        endpoint_url=WB_CONFIG_ENDPOINT.format(pk=gv_addon_pk),
+        requesting_user=requesting_user,
+        requested_resource=requested_resource
+    )
+
+
+def get_gv_result(**kwargs):  # -> JSONAPIResultEntry
+    '''Processes the result of a request to a GravyValet detail endpoint into a single JSONAPIResultEntry.
+
+    kwargs must match _make_gv_request
```

Comment:
i'm not crazy about obfuscating a call signature like this, but it's not _not_ pythonic...

Reply:
I just got sick of typing the same signature over and over again -- will spell it out